### PR TITLE
pipgd_polarization function, documentation and tests

### PR DESCRIPTION
**Pull request by RossanaTat**

### CODE REVIEW

**Comment by zander-prinsloo** [comment]
@randrescastaneda , the gd polarization functions, tests, and examples have been added by @RossanaTat. 
As discussed, the p0 argument is set to 0.5, gini is calculated internally if it is not supplied as an argument, and the dcm is determined internally before making use of newly exported `wbpip::gd_compute_polarization_l*` functions. Documentation has been produced, tests are passing, as are all `checks(vignettes = F)`.

**Review by zander-prinsloo — APPROVED** [review]

File: R/pipgd_dist.R [hunk 1]
```diff
@@ -529,8 +529,127 @@ pipgd_mld <- function(
...
+  } else {
+    match.arg(lorenz, c("lq", "lb"))
+  }
+
+  #   _________________________________________________________________
+  #   Set arguments
+  #   _________________________________________________________________
+
+  # Gini
+  if (is.null(gini)) {
```
Comment (randrescastaneda) [inline R/pipgd_dist.R]: Hi @zander-prinsloo, 

The `gini` argument should be available only for the users who want to provide their own gini. In principle, it should be available in the `params` object. Thus, to estimate the gini with `pipgd_gini` you need to check both, that `gini` argument is null and that the gini is not available either in `params`.

File: R/pipgd_dist.R [hunk 2]
```diff
@@ -529,8 +529,127 @@ pipgd_mld <- function(
...
+#' # Example with complete output
+#' pipgd_polarization(welfare  = pip_gd$L,
+#'                    weight   = pip_gd$P,
+#'                    complete = TRUE)
+#'
+pipgd_polarization <- function(
+    params     = NULL,
+    welfare    = NULL,
+    weight     = NULL,
+    mean       = 1,
```
Comment (randrescastaneda) [inline R/pipgd_dist.R:575]: As with my comment about the gini, we must make sure that the `mean` is included in params in previous calculations. If the user did not specify the mean, then it should be `mean = params$mean` or something like that. In `pipgd_params` the argument should be `mean = ` as it is in this polarization function.

**Review by randrescastaneda — CHANGES_REQUESTED** [review]
Hey @zander-prinsloo  and @RossanaTat , 

Please, address my comments on this function and also make sure that they are implemented throughout all the `pipgd` functions. 

Thanks.

**Pull request merged**